crates/gateway-client/src/lib.rs

```diff
@@ -227,22 +193,14 @@ impl<T: GatewayApi + Sync + Send> GatewayApi for std::sync::Arc<T> {
 /// where `N` is the consecutive retry iteration number `{1, 2, ...}`.
 #[derive(Debug, Clone)]
 pub struct Client {
-    /// This client is internally refcounted
     inner: reqwest::Client,
-    /// Starknet gateway URL.
     gateway: Url,
-    /// Starknet feeder gateway URL.
     feeder_gateway: Url,
+    is_retry_enabled: bool,
```

Comment:
Maybe an alternative is to attach the ability to retry to each query via a builder or something? 

As in, allow the default retry behaviour to be specified in the client as you have it here, to be used in most scenarios3

But then in specific situations, where the code **must** have a specific retry behaviour we allow overriding the client default by adding a `.retry(bool)` extension? I'm not entirely sure how one would do that, but to end with:
```rust
client
  .state_update(block_id)
  .retry(true)
  .await
```
one would probably need a trait extension on the Client API trait methods?

Reply:
Actually probably already possible as:
```rust
client
    .clone()
    .with_retry(bool)
    ....
```